src/main/java/org/galatea/starter/utils/DateHelpers.java

```diff
@@ -0,0 +1,66 @@
+package org.galatea.starter.utils;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import org.galatea.starter.domain.IexHistoricalPriceDTO;
+
+public class DateHelpers {
+  public static List<LocalDate> getDates(String range) {
+    LocalDate end = LocalDate.now();
+    List<LocalDate> totalDates = new ArrayList<>();
+    LocalDate start = end.minusDays(30); // default
+
+    // Reassign start value if a range was entered
+    if (range.equalsIgnoreCase("max")) {
+      start = end.minusYears(15);
+    } else if (range.equalsIgnoreCase("5y")) {
+      start = end.minusYears(5);
+    } else if (range.equalsIgnoreCase("2y")) {
+      start = end.minusYears(2);
+    } else if (range.equalsIgnoreCase("1y")) {
+      start = end.minusYears(1);
+    } else if (range.equalsIgnoreCase("ytd")) {
+      // to do
+    } else if (range.equalsIgnoreCase("6m")) {
+      start = end.minusMonths(6);
+    } else if(range.equalsIgnoreCase("3m")) {
+      start = end.minusMonths(3);
+    } else if(range.equalsIgnoreCase("1m") || range.equalsIgnoreCase("1mm")) {
+      start = end.minusMonths(1);
+    } else if(range.equalsIgnoreCase("5d")) {
+      start = end.minusDays(5);
+    }
+
+    while(!start.isAfter(end)) {
+      totalDates.add(start);
+      start = start.plusDays(1);
+    }
```

Comment:
Can you explain what this is for?

Reply:
This adds all the days within the range to a list. The loop ends when we are at the present day (end of range).

Reply:
The list is then used to get the first day in the range.